test/GraphQL.Conventions.Tests/Adapters/Engine/Types/CustomJsonGraphType.cs

```diff
@@ -1,14 +1,18 @@
 using System.Collections.Generic;
-using GraphQL.Language.AST;
 using GraphQL.Types;
+using GraphQLParser.AST;
 
 // ReSharper disable InconsistentNaming
 
 namespace Tests.Adapters.Engine.Types
 {
     #region .NET Class
-    public class JSON : ValueNode<IDictionary<string, object>>
+    public class JSON : GraphQLValue
```

Comment:
This test is completely wrong (not the fault of this PR).  Deserializing a GraphQL request cannot possibly deserialize to a `JSON` node.  Rather, `ParseLiteral` should be parsing a structured tree of objects and values, or interpreting a JSON string value stored as a string literal.  Further, the `ParseValue` method should probably just take the object and return it, as the GraphQL serializer should be deserializing variables into a structured map.  It won't ever be a `GraphQLValue` node of any kind.  And `ToAST` should convert the structured map into a tree of nodes.